### PR TITLE
[DROOLS-7000] class retention by JSONMashaller ObjectMapper._typeFact…

### PR DESCRIPTION
**Pull request by tkobayas**
…ory._typeCache

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7000

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
Jenkins retest this

**Comment by tkobayas** [comment]
GHA Java 8 : Maybe flaky
```
2022-06-02T07:37:30.7090841Z [ERROR] testDefaultLoadBalancerNoServersAvailable(org.kie.server.client.LoadBalancerClientTest)  Time elapsed: 5.182 s  <<< ERROR!
2022-06-02T07:37:30.7091425Z org.kie.server.common.rest.NoEndpointFoundException: No available endpoints found
2022-06-02T07:37:30.7108920Z    at org.kie.server.client.balancer.impl.AbstractBalancerStrategy.checkEmpty(AbstractBalancerStrategy.java:27)
```

**Comment by tkobayas** [comment]
Jenkins retest this

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java [hunk 1]
```diff
@@ -112,6 +113,9 @@ public class JSONMarshaller implements Marshaller {
 
     private boolean findDeserializerFirst = Boolean.parseBoolean(System.getProperty("org.kie.server.json.findDeserializerFirst.enabled", "true"));
 
+    private boolean typeFactoryPerMarshaller = Boolean.parseBoolean(System.getProperty("org.kie.server.json.typeFactoryPerMarshaller.enabled", "true"));
```
Comment (tkobayas) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java:116]: I wondered whether "true" or "false" is better as a default value.

Pros for "true" as default:
* Less memory retention especially for frequent deploy/dispose operations.

Pros for "false" as default:
* singleton `_typeFactory._typeCache` is a limited sized cache so it's not an infinite memory leak.
* Shared cache may reduce the heap size in some scenarios (e.g. many different deployments)

Considering the above, hmm, I vote for "true" as default.

**Comment by tkobayas** [comment]
Jenkins retest this

**Review by mariofusco — APPROVED** [review]

**Pull request merged**